### ISSUE TITLE
"Ouch"

### ISSUE THREAD

**Issue by maxnbk** [wimglenn/johnnydep#46]
reproducible with colour-science, something to do with the extras involved. Tried hacking the exception away and mucking with the following assertions but it got messy.

```
$ johnnydep colour-science
2019-12-10 18:37:06 [info     ] init johnnydist                [johnnydep.lib] dist=colour-science parent=None
Traceback (most recent call last):
  File "[...]/bin/johnnydep", line 8, in <module>
    sys.exit(main())
  File "[...]/site-packages/johnnydep/cli.py", line 62, in main
    print(dist.serialise(fields=args.fields, format=args.output_format, recurse=args.recurse))
  File "[...]/site-packages/johnnydep/lib.py", line 243, in serialise
    return tabulate.tabulate(table, headers="keys")
  File "[...]/site-packages/tabulate.py", line 1261, in tabulate
    list_of_lists, headers = _normalize_tabular_data(
  File "[...]/site-packages/tabulate.py", line 876, in _normalize_tabular_data
    rows = list(tabular_data)
  File "[...]/site-packages/johnnydep/lib.py", line 279, in gen_table
    for pre, _fill, node in anytree.RenderTree(johnnydist):
  File "[...]/site-packages/anytree/render.py", line 272, in __next
    children = node.children
  File "[...]/site-packages/johnnydep/lib.py", line 129, in children
    for dep in self.requires:
  File "[...]/site-packages/johnnydep/lib.py", line 96, in requires
    raise Exception("ouch")
Exception: ouch
```

**Comment by wimglenn** [comment]
@maxnbk  `colour-science` had a case that I hadn't encountered before (conditional dependencies specified across multiple "extras"). This should be fixed now, johnnydep v1.2 is released to PyPI.  [Here](http://dpaste.com/2B8H8GQ) is the huge dep tree with all extras opt-in.

**Comment by maxnbk** [comment]
Thanks @wimglenn , your fix appears to have addressed the problem I'm having. Much appreciated, and thanks for johnnydep in general
